scraper/scraper_bot.py

```diff
@@ -87,25 +93,26 @@ def download_and_convert_images_for_dataframe(df: pd.DataFrame) -> dict:
 def merge_datasets(old_df: pd.DataFrame, new_df: pd.DataFrame) -> pd.DataFrame:
     # Gather existing URLs from old_df using indices if it exists
     existing_image_urls = old_df['link'].tolist() if old_df is not None else []
-    
+
     print(f"Current rows: {len(existing_image_urls)}")
-    
+
     # Filter new_df to only include rows that aren't in old_df
     filtered_new_df = new_df.loc[~new_df['link'].isin(existing_image_urls)]
-    
+
     print(f"Rows to add: {filtered_new_df.shape[0]}")
 
     # Concatenate the old and filtered new dataframes
     merged_df = pd.concat([old_df, filtered_new_df]) if old_df is not None else filtered_new_df
-        
+
     return merged_df
 
 
-def get_latest_message_id(df: pd.DataFrame) -> str:
+def get_latest_message_id(df: pd.DataFrame, channel_id: str) -> str:
     try:
-        return df["message_id"].max()
+        # Find the largest message_id for the given channel_id
+        return df.loc[df['channel_id'] == channel_id]['message_id'].max()
```

Comment:
What is the channel_id check for? merging from different discords?

Reply:
I assumed for some reason that the datasets would be mixed. I'll adjust